### PR TITLE
update to infra refresher for OSP12

### PR DESCRIPTION
**Pull request by tzumainn**

### CODE REVIEW

File: app/models/manageiq/providers/openstack/infra_manager/refresh_parser.rb [hunk 1]
```diff
@@ -79,23 +79,39 @@ def ems_inv_to_hashes
...
-        @orchestration_service.list_resources(:stack => stack, :nested_depth => 2).body['resources']
+      def stack_resource_groups
+        return @stack_resource_groups if @stack_resource_groups
+        @stack_resource_groups = filter_stack_resources_by_resource_type(["OS::Heat::ResourceGroup"])
+      end
+
+      def stack_server_resource_types
+        return @stack_server_resource_types if @stack_server_resource_types
+        @stack_server_resource_types = ["OS::TripleO::Server", "OS::Nova::Server"]
+        @stack_server_resource_types += stack_resource_groups.map { |rg| "OS::TripleO::" + rg["resource_name"] + "Server" }
```
Comment (mansam) [inline app/models/manageiq/providers/openstack/infra_manager/refresh_parser.rb]: Is this nesting of Server under other types documented somewhere?

**Comment by mansam** [comment]
Thanks @tzumainn! Looks good, though I'm not well positioned to be able to test it. VCRs will just need to be updated :)

File: app/models/manageiq/providers/openstack/infra_manager/refresh_parser.rb (hunk 1 above)
Comment (tzumainn) [reply app/models/manageiq/providers/openstack/infra_manager/refresh_parser.rb]: hahahahahaha no, it's not

Comment (mansam) [reply app/models/manageiq/providers/openstack/infra_manager/refresh_parser.rb]: Hah, good job figuring it out then :)

**Comment by tzumainn** [comment]
@mansam oh, whoops. VCRs should be updated now!

**Comment by mansam** [comment]
awesome, looks good!

**Comment by aufi** [comment]
Maybe VCR re-record is needed..

**Comment by aufi** [comment]
Looks good to me, merging.

**Comment by simaishi** [comment]
Backported to Fine via https://github.com/ManageIQ/manageiq/pull/17101

**Pull request merged**